src/Microsoft.ComponentDetection.Orchestrator/Services/GraphTranslation/DefaultGraphTranslationService.cs

```diff
@@ -92,19 +92,22 @@ private IEnumerable<DetectedComponent> GatherSetOfDetectedComponentsUnmerged(IEn
                         // Return in a format that allows us to add the additional files for the components
                         var locations = dependencyGraph.GetAdditionalRelatedFiles();
 
-                        // graph authoritatively stores the location of the component
-                        locations.Add(location);
-
-                        foreach (var customLocation in componentCustomLocations)
+                        if (updateLocations)
```

Comment:
nit: can we add a comment here saying why locations shouldn't be updated? so we don't run into this question in the future